src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs

```diff
@@ -112,5 +114,22 @@ private void AddIfPrefixed(Dictionary<string, string> data, string key, string v
         }
 
         private static string NormalizeKey(string key) => key.Replace("__", ConfigurationPath.KeyDelimiter);
+
+        private static IDictionary CreateEnvironmentVariablesCache()
+        {
+            IDictionary environmentVariables;
+            s_environmentVariablesCache = environmentVariables = Environment.GetEnvironmentVariables();
```

Comment:
This will misbehave if someone adds/sets an environment variable before Gen2 runs (which can be anytime). I think a better fix would be to cache it inside Environment class for Windows too.

Reply:
> I think a better fix would be to cache it inside Environment class for Windows too.

If it was in `Environment` would it need to do a defensive copy on return since its returning an `IDictionary` that is writable; but unlinked from environment vars? That would mean if it was called only once it would create 2 `IDictionary`s which would be a regression; could alternatively make it a live `IDictionary` so changes are reflected in `Environment`, but that might be breaking?

> This will misbehave if someone adds/sets an environment variable before Gen2 runs 

Might be bigger issues there? 

* The `ConfigurationProvider` has a `Set` method; however `EnvironmentVariablesConfigurationProvider` doesn't call `Environment.SetEnvironmentVariable` if that set method is called but just updates that instance's copy.

* The `ConfigurationProvider` also has an `IChangeToken` which says to update and reload the configuration however if `Environment.SetEnvironmentVariable` is called it doesn't reload and fire that event (unlike with file config), nor is there currently an `Environment` event that it can listen to trigger that reload of configuration.